internal/collections/duration_collection.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package collections
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/corazawaf/coraza/v3/collection"
+	"github.com/corazawaf/coraza/v3/internal/corazarules"
+	"github.com/corazawaf/coraza/v3/internal/variables"
+	"github.com/corazawaf/coraza/v3/types"
+)
+
+type Duration struct {
```

Comment:
I really see this collection being problematic, first of all `Get` returns a different value every time, second `FindAll` returns a different value every time and nothing to do with `Get`. I wonder a. if this collection is a good example b. if implementing all methods make sense.

What is the use case for this? I would not include this in the core and instead wait for a use case.